modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/VersionComparator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (C) Copyright 2006-2020 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thierry Delprat
+ *     Anahide Tchertchian
+ */
+package org.nuxeo.apidoc.snapshot;
+
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Comparator for versions using 1 to 3 digits.
+ * <p>
+ * Makes sure version "11.2.3" is < "11.2.10" (so simple String ordering cannot be used).
+ *
+ * @since 20.0.0
+ */
+public class VersionComparator implements Comparator<String> {
+
+    public static final Pattern VERSION_REGEX = Pattern.compile("^(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:-.*)?$",
+            Pattern.CASE_INSENSITIVE);
+
+    @Override
+    public int compare(String version1, String version2) {
```

Comment:
It may be simpler to just split/map things into two `long[]` then call `Arrays.compare`, which exists since Java 9 and does array lexicographic comparison.

Reply:
Using `Arrays.compare` is tempting, but if by "just split/map", you mean getting rid of the regexp, i'm afraid it won't handle the 11.2.3-RC and 11.2-SNAPSHOT use cases (that i should have included in unit tests...)

Reply:
Ok if you have -RC and -SNAPSHOT cases then it's different

Reply:
I added more use cases (and complicated the code, please let me know what you think)

Reply:
I'm wondering, what about using an existing library? Such as `ComparableVersion ` from `maven-artifact` or https://github.com/vdurmont/semver4j (if it covers all use cases)?

Reply:
semver4j looks nice indeed, i can have a try at it (unless we'd like to stick to the maven logics?)

Reply:
Note i complicated the code to make sure "11.2" comes after "11.2.*" as well as "11.2-RC".
It occurred to me that on the contrary we may want to have "11.2-HF*" after "11.2", so it depends on our future versioning strategy (and if it does comply with semantic versioning as well as maven versioning)

Reply:
We expect to not have `-HF*` version anymore, but standard versions, if it can help. 
We still need to check the impacts it may have.

Reply:
FYI semver4j considers that "`11`" and "`11.0`" are invalid versions.

Reply:
You can change the mode to allow those kind of versions: https://github.com/vdurmont/semver4j#the-semver-object

Reply:
Ah! I'll have another try then (and i will check the maven ordering too just to see)

Reply:
So current ordering with regexp gives:

`[11.2.0, 11.2.1-SNAPSHOT, 11.2.1, 11.2.3-I20200929, 11.2.3-SNAPSHOT, 11.2.3, 11.2.10, 11.2.49, 11.2.53, 11.2-RC1, 11.2-RC2, 11.2-RC10, 11.2-bar, 11.2-foo, 11.2, 11.3, 11.10]`

With maven ComparableVersion, this gives:

`[11.2-RC1, 11.2-RC2, 11.2-RC10, 11.2.0, 11.2, 11.2-bar, 11.2-foo, 11.2.1-SNAPSHOT, 11.2.1, 11.2.3-SNAPSHOT, 11.2.3, 11.2.3-I20200929, 11.2.10, 11.2.49, 11.2.53, 11.3, 11.10]`

and with semver4j:

`[11.2-bar, 11.2-foo, 11.2-RC1, 11.2-RC10, 11.2-RC2, 11.2, 11.2.0, 11.2.1-SNAPSHOT, 11.2.1, 11.2.3-I20200929, 11.2.3-SNAPSHOT, 11.2.3, 11.2.10, 11.2.49, 11.2.53, 11.3, 11.10]`

so i guess i should keep current logics?

Reply:
Keeping current logics for now. Please note that this is used for display, as well as "latest" version retrieval logics.